################################################################################
SHORT DESCRIPTION: 
################################################################################
Avoid panic in cpufreq_gov_stop

################################################################################
LONG DESCRIPTION: 
################################################################################
The problem is that cpufreq_gov_stop is called without having a start first so
this patch will check if the givernor has been enabled before disabling it.

################################################################################
CHANGELOG 
################################################################################

################################################################################
REMOVAL 
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################
A rewrite of this patch has been submitted to upstream Xen by Christopher Clark
and accepted in upstream commit: e7ec6f5f32cd2d0f723083cde3d7761c4e675f2c

This patch can be removed from OpenXT in the next major version upgrade of Xen.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################

################################################################################
PATCHES 
################################################################################
Index: xen-4.9.0/xen/drivers/cpufreq/cpufreq_ondemand.c
===================================================================
--- xen-4.9.0.orig/xen/drivers/cpufreq/cpufreq_ondemand.c
+++ xen-4.9.0/xen/drivers/cpufreq/cpufreq_ondemand.c
@@ -273,8 +273,11 @@ int cpufreq_governor_dbs(struct cpufreq_
         break;
 
     case CPUFREQ_GOV_STOP:
-        dbs_timer_exit(this_dbs_info);
-        dbs_enable--;
+        if (this_dbs_info->enable)
+        {
+            dbs_timer_exit(this_dbs_info);
+            dbs_enable--;
+        }
 
         break;
 
